src/Sulu/Bundle/WebsiteBundle/Twig/Content/ContentPathTwigExtension.php

```diff
@@ -66,41 +58,55 @@ public function getFunctions()
     /**
      * {@inheritdoc}
      */
-    public function getContentPath($url, $webspaceKey = null, $locale = null, $domain = null, $scheme = null)
+    public function getContentPath($route, $webspaceKey = null, $locale = null, $domain = null, $scheme = null)
     {
         if (!$this->requestAnalyzer) {
-            return $url;
+            return $route;
         }
 
-        $domain = $domain ?: $this->requestAnalyzer->getAttribute('host');
         $scheme = $scheme ?: $this->requestAnalyzer->getAttribute('scheme');
         $locale = $locale ?: $this->requestAnalyzer->getCurrentLocalization()->getLocale();
 
         if ($webspaceKey !== null) {
-            if (!$this->webspaceManager->findWebspaceByKey($webspaceKey)->hasDomain($domain, $this->environment)) {
+            if (null !== $domain
+                && !$this->webspaceManager->findWebspaceByKey($webspaceKey)->hasDomain($domain, $this->environment)
+            ) {
                 $domain = null;
```

Comment:
Isn't that some kind of hiding bugs from the developer? Wouldn't throwing an exception here be more appropriate? This way the developer might just wonder why the domain parameter is ignored.